### ISSUE TITLE
Style/EmptyLiteral is suggesting inequivalent replacements! 

### ISSUE THREAD

**Issue by ellcs** [rubocop-hq/rubocop#7328]
The cop `Style/EmptyLiteral` is telling to replace `String.new` with a string literal `''`. However we figured out, that this is not equivalent! The difference:

* Creating a string as a literal `''` sets the encoding to `UTF-8`. 
* Using `String.new` sets the encoding to `ASCII-8BIT`.

```ruby
irb(main):001:0> ''.encoding
=> #<Encoding:UTF-8>
irb(main):002:0> String.new.encoding
=> #<Encoding:ASCII-8BIT>
```

--------
## Expected behavior
 Use string literal `''.force_encoding(Encoding::ASCII_8BIT)` instead of `String.new`.
## Actual behavior
 Use string literal `''` instead of `String.new`.

## Steps to reproduce the problem

```bash
$ echo "str = String.new; puts str.encoding" > file.rb
$ ruby file.rb
ASCII-8BIT
$ rubocop --only EmptyLiteral --auto-correct file.rb
Inspecting 1 file
C

Offenses:

file.rb:1:7: C: [Corrected] Style/EmptyLiteral: Use string literal '' instead of String.new.
str = String.new; puts str.encoding
      ^^^^^^^^^^

1 file inspected, 1 offense detected, 1 offense corrected
$ ruby file.rb
UTF-8
```

## rubocop version
```
$ rubocop --version
0.57.2
```

**Comment by koic** [comment]
I remembered that I investigated this problem before.
https://github.com/rails/rails/pull/29506#issuecomment-309685349

What problem use case do you have by this auto-correct?

OTOH, the auto-correct for `Array.new` and `Hash.new` are fine.
It would be better to provide an option that strictly warning only argumentless `String.new` over `SafeAutoCorrect: false` if this issue is addressed. And I'm concerned that `String.new` has a potential problem. I think it's usually auto-corrected by `""` literal, so I think this option can enable warning by default.

**Comment by koic** [comment]
Another idea.
`''.force_encoding(Encoding::ASCII_8BIT)` is strictly correct. However, I'm worried about being redundant if ASCII_8BIT encoding is not expected.
It may be better to provide an encoding compatible option.

**Comment by ellcs** [comment]
Hey kioc! Thank you for your fast reply! 

> I remembered that I investigated this problem before.
> [rails/rails#29506 (comment)](https://github.com/rails/rails/pull/29506#issuecomment-309685349)
> 
> What problem use case do you have by this auto-correct?
> 
> OTOH, the auto-correct for `Array.new` and `Hash.new` are fine.
> It would be better to provide an option that strictly warning only argumentless `String.new` over `SafeAutoCorrect: false` if this issue is addressed. And I'm concerned that `String.new` has a potential problem. I think it's usually auto-corrected by `""` literal, so I think this option can enable warning by default.

I work with different external processes. They do use different encodings for stdin and stdout. That led to displaying issues within the gui. So i do use `ASCII_8BIT` in my application core, as convention. The buffer was initialized with `String.new`, for parsing a process' stdout. 

> Another idea.
> `''.force_encoding(Encoding::ASCII_8BIT)` is strictly correct. 

So your two suggestions are:
- Change behavior for `String.new`, but not `String.new('i already have an encoding')`. Empty strings only.
- Disable [SafeAutoCorrect](https://github.com/rubocop-hq/rubocop/blob/master/manual/auto_correct.md#safe-auto-correct)

Both suggestions seem good to me. 

> However, I'm worried about being redundant if ASCII_8BIT encoding is not expected.
> It may be better to provide an encoding compatible option.

I see your point. People might have the wrong assumption. And yes, it is rubocops job to show that. For me it is more important that i receive message that there might be encoding issues, when using the cop.

But when suggesting a replacement, it should be 100% equivalent. No matter what the programmer wrote.
